internal/planner/select.go

```diff
@@ -255,19 +255,24 @@ func (n *selectNode) initSource() ([]aggregateNode, error) {
 		origScan.filter = n.filter
 		n.filter = nil
 
-		// If we have both a DocID and a CID, then we need to run
-		// a TimeTravel (History-Traversing Versioned) query, which means
-		// we need to propagate the values to the underlying VersionedFetcher
+		// If we have a CID, then we need to run a TimeTravel (History-Traversing Versioned)
+		// query, which means we need to propagate the values to the underlying VersionedFetcher
 		if n.selectReq.Cid.HasValue() {
 			c, err := cid.Decode(n.selectReq.Cid.Value())
 			if err != nil {
 				return nil, err
 			}
+
+			var docID string
+			if len(n.selectReq.DocIDs.Value()) > 0 {
+				docID = n.selectReq.DocIDs.Value()[0]
```

Comment:
thought: Passing the docID for a timetravel query is unnecessary as it will never be used. Even worst, it's confusing the reader by pretending to be useful for the query. One can pass an invalid docID and still get a valid response. We should probably remove references to docID in this block.

tought: It might be more appropriate to allow setting CID or docID but not both in the query params.

Reply:
Oh I forgot to test that, sorry.

There is a small chance it is still used within the planner nodes, and I actually want it to be so, so that a docID can be provided alongside a collection cid - otherwise there is a gap in the query syntax like you said.  IIRC John really likes being able to set the `docID` and `cid` in the same query, even though the `docID` has always been somewhat pointless from an external perspective.

That said, even then the docID in `Spans` is almost certainly irrelevant (I didn't notice I'd removed it completely from the versionedFetcher) - and so this block deserves a revisit for being highly misleading like you said.

I'll have a play, if it looks like it'll take a bit more effort to get collection-cids with docIDs working I might add a temp error and we can add support whenever it becomes high enough priority.

- [ ] col-cid/docID interplay stuff

Reply:
It worked with the prod code unchanged, test is `TestQuerySimpleWithCidOfBranchableCollectionAndDocID`.

I think I remember actually thinking this through, and saw that the span is passed on to the backing document fetcher.  But looking through the code I don't see that atm.  It might be that the split logic refuses to split the filter as the first fetcher is not a doc-fetcher, leaving the span/select node to handle the docID filter.

Where this happens is not really important now though, I'll see if I can remove the block here without stuff breaking :)

- [ ] Remove docID from prefix (it isn't actually a prefix either)

Reply:
Removing it was problem free. I've documented the oddness here, and will merge this PR once the CI is done.